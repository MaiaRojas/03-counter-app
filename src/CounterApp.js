//import React, { Fragment } from 'react';
import React, { useState } from 'react';
import PropTypes from 'prop-types';


//Fc => Functional components
const CounterApp = ({ value = 10 }) => {

  const [counter, setcounter] = useState(value);
  //handleAdd
  const handleAdd = () => setcounter(counter + 1);
  const handleReset = () => setcounter(value);
  const handleSub = () => setcounter(counter - 1);

  return <>
    <h1>CounterApp</h1>
    <p>{counter}</p>
    <button onClick={handleAdd} value='+1'>+1</button>
    <button onClick={handleReset} value='reset'>Reset</button>
    <button onClick={handleSub} value='-1'>-1</button>
  </>

};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;