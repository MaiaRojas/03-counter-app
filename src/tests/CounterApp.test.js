import { shallow } from "enzyme";
import React from "react";
import CounterApp from "../CounterApp";


describe('Pruebas en <CounterApp />', () => {
  let wrapper = shallow(<CounterApp value={10} />);
  beforeEach(() => {
    wrapper = shallow(<CounterApp value={10} />);
  });
  

  test('should  be show <CounterApp/> right', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should  be show default value 100', () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const valueCounter = wrapper.find('p').text().trim();

    expect(valueCounter).toBe('100');
  });

  test('Debe incrementar con el boton +1', () => {
    wrapper.find('button').at(0).simulate('click');
    const valueCounter = wrapper.find('p').text().trim();

    expect(valueCounter).toBe('11');
  });

  test('Debe decrementar con el boton -1', () => {
    wrapper.find('button').at(2).simulate('click');
    const valueCounter = wrapper.find('p').text().trim();

    expect(valueCounter).toBe('9');
  });

  test('Debe colocar el value default con el btn reset', () => {
    wrapper = shallow(<CounterApp value={110} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');

    const valueCounter = wrapper.find('p').text().trim();

    expect(valueCounter).toBe('110');
  });
  
})