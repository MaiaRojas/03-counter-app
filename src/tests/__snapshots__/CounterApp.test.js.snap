// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> should  be show <CounterApp/> right 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
    value="+1"
  >
    +1
  </button>
  <button
    onClick={[Function]}
    value="reset"
  >
    Reset
  </button>
  <button
    onClick={[Function]}
    value="-1"
  >
    -1
  </button>
</Fragment>
`;
